### PR TITLE
Selected series

### PR DESCRIPTION
**Pull request by wissog**
AR-754

### CODE REVIEW

File: public/locales/en/common.json [hunk 1]
```diff
@@ -592,4 +596,4 @@
     "copy-text": "Copy text",
     "disable-share": "Cancel"
   }
-}
+}
```
Comment (gshilin) [inline public/locales/en/common.json:599]: Return <new line> here please

File: src/components/Sections/Lessons/tabs/Series/Container.js [hunk 2]
```diff
@@ -1,63 +1,193 @@
...
+      return null;
+    }
+
+    const { id, children, name } = node;
+
+    // expand all levels excluding the leafs
+    const expandedLevel = !isEmpty(children) && children.some(ch => !isEmpty(ch.children));
+    const showExpandButton = !expandedLevel && children?.length > visibleItemsCount;
+    const expanded = expandedNodes.has(id);
+    const key = id + '#' + level;
```
Comment (gshilin) [inline src/components/Sections/Lessons/tabs/Series/Container.js:63]: key = `${id}#${level}`

File: src/components/Sections/Lessons/tabs/Series/Container.js [hunk 3]
```diff
@@ -1,63 +1,193 @@
...
+  const getHeader = node => node.id === "byTopics"
+    ? t(`lessons.tabs.seriesTree.byTopics`)
+    : t(`lessons.tabs.seriesTree.bySourcesOf`) + ' ' + node.name;
+
+  const renderKabbalist = node => {
+    if (node?.children?.length > 0){
+      const initLevel = 2;
+
+      return (
+        <div key={node.id + '#' + initLevel}>
```
Comment (gshilin) [inline src/components/Sections/Lessons/tabs/Series/Container.js:106]: use ` please

File: src/components/Sections/Lessons/tabs/Series/Container.js [hunk 4]
```diff
@@ -1,63 +1,193 @@
...
+      ? k1_index - k2_index
+      : k1_index > k2_index
+        ? k1_index
+        : k2_index;
+  }
+
+  const renderGrid = () => (
+    <Grid columns={3} padded>
+      <Grid.Row>
+        <Grid.Column key='bs#1' className="topics__section">
```
Comment (gshilin) [inline src/components/Sections/Lessons/tabs/Series/Container.js:141]: what do you need constant key for?

File: src/components/Sections/Lessons/tabs/Series/Container.js [hunk 5]
```diff
@@ -1,63 +1,193 @@
...
+          {renderKabbalist(dataTree.find(node => node.id === 'bs'))}
+        </Grid.Column>
+        <Grid.Column key='various#1' className="topics__section">
```
Comment (gshilin) [inline src/components/Sections/Lessons/tabs/Series/Container.js:144]: what do you need constant key for?

File: src/components/Sections/Lessons/tabs/Series/Container.js [hunk 6]
```diff
@@ -1,63 +1,193 @@
...
+          {
+            dataTree
+              .filter(node => node.id !== 'bs' && node.id !== 'byTopics')
+              .sort(sortKabbalists)
+              .map(renderKabbalist)
+          }
+        </Grid.Column>
+        <Grid.Column key='byTopics#1' className="topics__section">
```
Comment (gshilin) [inline src/components/Sections/Lessons/tabs/Series/Container.js:152]: what do you need constant key for?

File: src/components/Sections/Topics/TopicContainer.js [hunk 7]
```diff
@@ -24,299 +23,263 @@ import {
...
-  },
-  dispatch => bindActionCreators({
-    fetchStats: topicsActions.fetchStats
-  }, dispatch)
-)(withNamespaces()(TopicContainer));
+TopicContainer.propTypes = {
+  t: PropTypes.func.isRequired,
+};
+
+export default withNamespaces()(TopicContainer);
```
Comment (gshilin) [inline src/components/Sections/Topics/TopicContainer.js:285]: return new line

File: src/redux/modules/lessons.js [hunk 8]
```diff
@@ -118,81 +118,187 @@ export const reducer = handleActions({
...
 
 export const selectors = {
   getWip,
   getErrors,
   getLecturesByType,
-  getSeriesBySource,
-};
+  getSeriesIDs,
+  getSeriesTree,
+};
```
Comment (gshilin) [inline src/redux/modules/lessons.js:304]: return it back

**Review by gshilin — CHANGES_REQUESTED** [review]

File: public/locales/en/common.json (hunk 1 above)
Comment (wissog) [reply public/locales/en/common.json:599]: it's a json... what to return?

File: src/components/Sections/Lessons/tabs/Series/Container.js (hunk 4 above)
Comment (wissog) [reply src/components/Sections/Lessons/tabs/Series/Container.js:141]: had errors on duplicate keys, so i made

File: src/components/Sections/Lessons/tabs/Series/Container.js (hunk 3 above)
Comment (wissog) [reply src/components/Sections/Lessons/tabs/Series/Container.js:106]: how it is better

File: src/components/Sections/Lessons/tabs/Series/Container.js (hunk 5 above)
Comment (wissog) [reply src/components/Sections/Lessons/tabs/Series/Container.js:144]: duplicate key errors

**Review by gshilin — APPROVED** [review]

**Pull request merged**